src/services/forgot-password.ts

```diff
@@ -0,0 +1,28 @@
+import "server-only";
+
+import { createHash } from "crypto";
+import { and, eq } from "drizzle-orm";
+
+import { FORGOT_PASSWORD_TOKEN_EXPIRY_TIME } from "@/lib/constants";
+import { db } from "@/lib/db";
+import { forgotPasswordTokens } from "@/lib/db/schema";
+
+/**
+ * creates and inserts a 6-digit verification token into the database
+ *
+ */
+export const insertForgotPasswordToken = async (email: string) => {
+  await db
+    .delete(forgotPasswordTokens)
+    .where(and(eq(forgotPasswordTokens.identifier, email)));
+
+  const hashedToken = createHash('sha256').update(email + (new Date()).toString()).digest('hex');
```

Comment:
actually this isn't very secure -- if you know the email if the user and the approximate time which they made the request the token can easily be brute forced with a script.

we should include an additional salt, using a secret code that is only known to use (in the .env)

jwt meets many of the requirements here, and I think might be better for this use case.
https://github.com/formbricks/formbricks/blob/main/packages/lib/jwt.ts